### ISSUE TITLE
Downgrade from v0.43.0 to v0.42.0

### ISSUE THREAD

**Issue by rezaebrahimi1** [tektoncd/pipeline#5955]
# Expected Behavior
Pipelines can be visible after running **# kubectl get pipeline** command.
# Actual Behavior
**# kubectl get pipeline** says there are no king pipeline in the cluster.
# Steps to Reproduce the Problem
1. Install tekton-pipeline latest version (currently v0.43.0) 
2. Without uninstalling deployed tekton-pipeline in step 1 (v0.43.0), deploy v0.42.0 on the same cluster.
3. Run **# kubectl get pipeline** command

# Additional Info

- Kubernetes version: 1.24

  **Output of `kubectl version`:**

  ```
  (paste your output here)
  ```

- Tekton Pipeline version:

  **Output of `tkn version` or `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`**

**Comment by vdemeester** [comment]
Hey @rezaebrahimi1,

Usually, we don't really support "downgrades", especially between versions that do not necessarily serve the same CRD versions. 

That said, I *think* this might be due to the fact that you didn't "uninstall" 0.43, aka not removing `v1` CRDs. What most likely happens here is that `kubectl get pipeline` will get the latest version of the CRD which is, in your case, on step 3, `v1`. That `v1` CRD is not handled *at all* by the 0.42 controller, nor the conversion webhook. This means, there is not gonna be any conversion from `v1beta1` to `v1` happening, and thus the list will just be plain empty.

To make sure that's it, can you try doing a `kubectl get Pipeline.v1beta1.tekton.dev` ? I feel it would give you some results. `kubectl get Pipeline.v1.tekton.dev` on the other hand would return you an empty list (because there is *no* stored `v1` types nor automatic conversion happening, but the CRD is still present).

From my point of view, it's definitely not a bug though 😅 , more an "unspported" case.

**Comment by pritidesai** [comment]
Hey @rezaebrahimi1 wondering why are you deploying pipelines `0.43` followed by `0.42`?

**Comment by rezaebrahimi1** [comment]
> Hey @rezaebrahimi1 wondering why are you deploying pipelines `0.43` followed by `0.42`?

Hi, the reason is
I had a pipeline (the first task of pipeline was git-clone) and that worked correctly with v0.42.0,
I give it to our QA team to do some tests,
they ran pipeline with version v0.43.0 and they faced with following error when pipeline was executing git-clone pod:
**Error executing command: fork/exec /tekton/scripts/script-0- exec format error**.
So i downgrade to v0.42.0 and i encounter with mentioned unsupported case :) (as @vdemeester said)
Uninstalling v0.43.0 and installing v0.42.0 resolved the problem for me.

**Comment by rezaebrahimi1** [comment]
> Hey @rezaebrahimi1,
> 
> Usually, we don't really support "downgrades", especially between versions that do not necessarily serve the same CRD versions.
> 
> That said, I _think_ this might be due to the fact that you didn't "uninstall" 0.43, aka not removing `v1` CRDs. What most likely happens here is that `kubectl get pipeline` will get the latest version of the CRD which is, in your case, on step 3, `v1`. That `v1` CRD is not handled _at all_ by the 0.42 controller, nor the conversion webhook. This means, there is not gonna be any conversion from `v1beta1` to `v1` happening, and thus the list will just be plain empty.
> 
> To make sure that's it, can you try doing a `kubectl get Pipeline.v1beta1.tekton.dev` ? I feel it would give you some results. `kubectl get Pipeline.v1.tekton.dev` on the other hand would return you an empty list (because there is _no_ stored `v1` types nor automatic conversion happening, but the CRD is still present).
> 
> From my point of view, it's definitely not a bug though sweat_smile , more an "unspported" case.

I will test it when our new test cluster be prepared.
thanks

**Comment by vdemeester** [comment]
> **Error executing command: fork/exec /tekton/scripts/script-0- exec format error**.

This error usually means the binary that is "executed" doesn't correspond to the node/container 's architecture. Are both cluster using the same arch (x86_64, or arm64, or …). If yes, then it might be the image reference or something that is weird. It seems really weird that 0.43 would bring an error like this on its ownk

**Comment by rezaebrahimi1** [comment]
> > **Error executing command: fork/exec /tekton/scripts/script-0- exec format error**.
> 
> This error usually means the binary that is "executed" doesn't correspond to the node/container 's architecture. Are both cluster using the same arch (x86_64, or arm64, or …). If yes, then it might be the image reference or something that is weird. It seems really weird that 0.43 would bring an error like this on its ownk

Yes they were on both clusters. I will test it today on our new test cluster (deployed by kubespray)
